pkg/apis/troubleshoot/v1beta2/hostcollector_shared.go

```diff
@@ -73,13 +73,38 @@ type TCPConnect struct {
 	Timeout           string `json:"timeout,omitempty"`
 }
 
+// FilesystemPerformance benchmarks sequential write latency on a single file.
+// The optional background IOPS feature attempts to mimic real-world conditions by running read and
+// write workloads prior to and during benchmark execution.
 type FilesystemPerformance struct {
-	HostCollectorMeta  `json:",inline" yaml:",inline"`
+	HostCollectorMeta `json:",inline" yaml:",inline"`
+	// The size of each write operation performed while benchmarking. This does not apply to the
+	// background IOPS feature if enabled, since those must be fixed at 4096.
 	OperationSizeBytes uint64 `json:"operationSize,omitempty"`
-	Directory          string `json:"directory,omitempty"`
-	FileSize           string `json:"fileSize,omitempty"`
-	Sync               bool   `json:"sync,omitempty"`
-	Datasync           bool   `json:"datasync,omitempty"`
+	// The directory where the benchmark will create files.
+	Directory string `json:"directory,omitempty"`
+	// The size of the file used in the benchmark. The number of IO operations for the benchmark
+	// will be FileSize / OperationSizeBytes. Accepts valid Kubernetes resource units such as Mi.
+	FileSize string `json:"fileSize,omitempty"`
+	// Whether to call sync on the file after each write. Does not apply to background IOPS task.
+	Sync bool `json:"sync,omitempty"`
+	// Whether to call datasync on the file after each write. Skipped if Sync is also true.
+	Datasync bool `json:"datasync,omitempty"`
+
+	// Enable the background IOPS feature.
+	EnableBackgroundIOPS bool `json:"enableBackgroundIOPS"`
+	// How long to run the background IOPS read and write workloads prior to starting the benchmarks.
+	BackgroundIOPSWarmupSeconds int `json:"backgroundIOPSWarmupSeconds"`
+	// The target write IOPS to run while benchmarking. This is a limit and there is no guarantee
+	// it will be reached.
+	BackgroundWriteIOPS int `json:"backgroundWriteIOPS"`
+	// The target read IOPS to run while benchmarking. This is a limit and there is no guarantee
+	// it will be reached.
+	BackgroundReadIOPS int `json:"backgroundReadIOPS"`
+	// Number of threads to use for background write IOPS.
+	BackgroundWriteIOPSJobs int `json:"backgroundWriteIOPSJobs"`
+	// Number of threads to use for background read IOPS.
+	BackgroundReadIOPSJobs int `json:"backgroundReadIOPSJobs"`
```

Comment:
are iops split over the threads, or per-thread?

Reply:
Total for all threads. I'll update the comment.